internal/kubelet/client.go

```diff
@@ -69,6 +77,46 @@ type ClientProvider interface {
 	BuildClient() (Client, error)
 }
 
+type kubeConfigClientProvider struct {
+	endpoint string
+	cfg      *ClientConfig
+	logger   *zap.Logger
+}
+
+func (p *kubeConfigClientProvider) BuildClient() (Client, error) {
+	if err := p.cfg.APIConfig.Validate(); err != nil {
+		return nil, err
+	}
```

Comment:
This is not needed. Validation is called by the component builder automatically if the `Validate` is defined